connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultInboundConnectorContextFactory.java

```diff
@@ -0,0 +1,102 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.inbound;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
+import io.camunda.connector.api.validation.ValidationProvider;
+import io.camunda.connector.feel.FeelEngineWrapper;
+import io.camunda.connector.runtime.core.inbound.correlation.InboundCorrelationHandler;
+import io.camunda.connector.runtime.core.secret.SecretProviderAggregator;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+public class DefaultInboundConnectorContextFactory implements InboundConnectorContextFactory {
+  private final ObjectMapper objectMapper;
+  private final InboundCorrelationHandler correlationHandler;
+  private final SecretProviderAggregator secretProviderAggregator;
+  private final ValidationProvider validationProvider;
+  private final OperateClientAdapter operateClientAdapter;
+  private final FeelEngineWrapper feelEngineWrapper;
+
+  public DefaultInboundConnectorContextFactory(
+      final ObjectMapper mapper,
+      final InboundCorrelationHandler correlationHandler,
+      final SecretProviderAggregator secretProviderAggregator,
+      final ValidationProvider validationProvider,
+      final OperateClientAdapter operateClientAdapter,
+      final FeelEngineWrapper feelEngineWrapper) {
+    this.objectMapper = mapper;
+    this.correlationHandler = correlationHandler;
+    this.secretProviderAggregator = secretProviderAggregator;
+    this.validationProvider = validationProvider;
+    this.operateClientAdapter = operateClientAdapter;
+    this.feelEngineWrapper = feelEngineWrapper;
+  }
+
+  @Override
+  public <T extends InboundConnectorExecutable<?>> InboundConnectorContext createContext(
+      final InboundConnectorDefinitionImpl newConnector,
+      final Consumer<Throwable> cancellationCallback,
+      final Class<T> executableClass) {
+
+    InboundConnectorContext inboundContext =
+        new InboundConnectorContextImpl(
+            secretProviderAggregator,
+            validationProvider,
+            newConnector,
+            correlationHandler,
+            cancellationCallback,
+            objectMapper);
+
+    if (isIntermediateContext(executableClass)) {
+      inboundContext =
+          new InboundIntermediateConnectorContextImpl(
+              inboundContext,
+              operateClientAdapter,
+              validationProvider,
+              feelEngineWrapper,
+              objectMapper);
+    }
```

Comment:
```
if (isIntermediateContext(executableClass)) {
...
else {
inboundContext =
        new InboundConnectorContextImpl(
            secretProviderAggregator,
            validationProvider,
            newConnector,
            correlationHandler,
            cancellationCallback,
            objectMapper);
}
```

Lets try to avoid object creation if not needed.

Reply:
Ah, sorry. Didnt see that. Works for me.